### ISSUE TITLE
Job time is possibly accounting for the full wrapper time

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#573]
In GitLab by @mcastril on Aug 14, 2020, 12:36

I have got 2 similar experiments. In one of them the time reported for the last SIM is too high. I think that it maybe is counting all the wrapper time instead of just the SIM time.

http://bscesweb04.bsc.es/autosubmitapp/experiment/a31w

http://bscesweb04.bsc.es/autosubmitapp/experiment/a31x

**Comment by kinow** [comment]
In GitLab by @mcastril on Aug 14, 2020, 12:45

In fact the time reported by the Userportal for the job (wrapper) is 1 day, 8 hours, 38 minutes

**Comment by kinow** [comment]
In GitLab by @wuruchi on Aug 14, 2020, 12:47

Hello @mcastril 

When a job finishes, a script is executed on SLURM to retrieve the times for that JobId.

Can you execute `sacct -n -j {0} -o JobId%20,State,Submit,Start,End,ConsumedEnergy,MaxRSS,AveRSS` where `{0}` is the id of your job on slurm.

**Comment by kinow** [comment]
In GitLab by @mcastril on Aug 14, 2020, 12:49

Sure:

```
bsc32252@login2:~> sacct -n -j 11303659 -o JobId%20,State,Submit,Start,End,ConsumedEnergy,MaxRSS,AveRSS
            11303659  COMPLETED 2020-08-12T21:25:48 2020-08-12T22:07:31 2020-08-14T06:45:31          4.02G                       
      11303659.batch  COMPLETED 2020-08-12T22:07:31 2020-08-12T22:07:31 2020-08-14T06:45:31         34.38M    213900K     18716K 
     11303659.extern  COMPLETED 2020-08-12T22:07:31 2020-08-12T22:07:31 2020-08-14T06:45:31          2.01G       814K     69.52K 
          11303659.0  COMPLETED 2020-08-12T22:07:50 2020-08-12T22:07:50 2020-08-13T04:39:09        402.69M  65214765K 31331064.+ 
          11303659.1  COMPLETED 2020-08-13T04:40:54 2020-08-13T04:40:54 2020-08-13T11:10:39        401.18M  64625179K  30468074K 
          11303659.2  COMPLETED 2020-08-13T11:12:50 2020-08-13T11:12:50 2020-08-13T17:42:59        401.75M  64402708K 31308270.+ 
          11303659.3  COMPLETED 2020-08-13T17:45:08 2020-08-13T17:45:08 2020-08-14T00:13:27        400.26M  65400964K 31370092.+ 
          11303659.4  COMPLETED 2020-08-14T00:15:25 2020-08-14T00:15:25 2020-08-14T06:43:51        400.26M  64579124K 32208449.+ 
```

**Comment by kinow** [comment]
In GitLab by @mcastril on Aug 14, 2020, 12:51

`2020-08-14T00:15:25 2020-08-14T06:43:51`

Looks ok right?

**Comment by kinow** [comment]
In GitLab by @mcastril on Aug 14, 2020, 12:52

This is the info for the "good" one, to compare to:

```
bsc32252@login2:~> sacct -n -j 11303562 -o JobId%20,State,Submit,Start,End,ConsumedEnergy,MaxRSS,AveRSS
            11303562  COMPLETED 2020-08-12T20:55:26 2020-08-12T21:54:54 2020-08-14T08:07:16          4.16G                       
      11303562.batch  COMPLETED 2020-08-12T21:54:54 2020-08-12T21:54:54 2020-08-14T08:07:16         38.56M    423232K     22344K 
     11303562.extern  COMPLETED 2020-08-12T21:54:54 2020-08-12T21:54:54 2020-08-14T08:07:16          2.08G       805K     70.08K 
          11303562.0  COMPLETED 2020-08-12T21:55:10 2020-08-12T21:55:10 2020-08-13T04:49:05        420.23M  66122628K 32163676.+ 
          11303562.1  COMPLETED 2020-08-13T04:50:39 2020-08-13T04:50:39 2020-08-13T11:33:16        410.73M  66533810K 30743576.+ 
          11303562.2  COMPLETED 2020-08-13T11:35:16 2020-08-13T11:35:16 2020-08-13T18:24:10        416.36M  65968156K 32465459.+ 
          11303562.3  COMPLETED 2020-08-13T18:26:20 2020-08-13T18:26:20 2020-08-14T01:24:44        424.15M  66063881K 32872684.+ 
          11303562.4  COMPLETED 2020-08-14T01:26:48 2020-08-14T01:26:48 2020-08-14T08:05:46        407.79M  66547556K 32010977.+ 
```

**Comment by kinow** [comment]
In GitLab by @wuruchi on Aug 14, 2020, 13:09

Thank you @mcastril 

Looks like the function is not handling well wrappers that have had a pause in their execution. Looking into it.

**Comment by kinow** [comment]
In GitLab by @mcastril on Aug 14, 2020, 13:16

> Looks like the function is not handling well wrappers that have had a pause in their execution. Looking into it.

I am curious about the pause. I think they run in a row, as they didn't have any external dependency.

**Comment by kinow** [comment]
In GitLab by @wuruchi on Aug 14, 2020, 17:53

Hello @mcastril 

I have identified the source of the problem. It seems that parsing the SLURM command output was trickier than expected. While I fix it, I have disabled the job historical database in the API, meaning that the GUI will show accurate times using the old centralized database that the workers on the server update periodically. You can check that the GUI is now showing correct running times.

**Comment by kinow** [comment]
In GitLab by @mcastril on Aug 14, 2020, 18:03

Thanks @wuruchi , indeed it is correct now.

**Comment by kinow** [comment]
In GitLab by @wuruchi on Aug 20, 2020, 10:44

Hello @mcastril 

After careful consideration and experimentation, here is the situation:

* The process of submitting wrappers and testing the output of the SLURM `sacct` command can be time-consuming and prone to error. Also, different types of wrappers use different submission configurations that can influence the output of the `sacct` command. In consequence, trying to completely process the output of `sacct` when we proceed to store the information in the database is not efficient.
* Then, Autosubmit can identify when the inner jobs of a wrapper have been completed. This gives us a more or less reliable source of information for the submit, start, finish times for jobs that belong to wrappers.
* In the case of jobs that do not belong to wrappers, the current method of time and energy information retrieval is reliable. However, most `SIM` jobs run in wrappers, so this method is not enough.

Here is the plan:

* Keep the current method for jobs not belonging to wrappers.
* For jobs in wrappers, store time information from the data generated by Autosubmit. Furthermore, when we detect that the wrapper has finished, either `COMPLETED` or `FAILED`, capture the `sacct` output and store it in the database.
* When we need to show job time and energy information for jobs in wrappers, we will use that `sacct` output and perform some pre-processing to determine the individual job values. In this way, future changes in wrappers should not require changes in the way information is stored, but only on the pre-processing function.

**Comment by kinow** [comment]
In GitLab by @wuruchi on Aug 21, 2020, 11:07

mentioned in commit a49dc06a3bff5606000eca2a370d0ee23bb9dc8e

**Comment by kinow** [comment]
In GitLab by @wuruchi on Aug 21, 2020, 11:07

mentioned in commit 623f79abd22ce440e687565ec253da5242ea7af2

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 7, 2020, 14:25

closed via merge request !191

**Comment by kinow** [comment]
In GitLab by @wuruchi on Sep 7, 2020, 14:25

closed via commit a49dc06a3bff5606000eca2a370d0ee23bb9dc8e
